### ISSUE TITLE
Authenticate with form parameters

### ISSUE THREAD

**Issue by jgadbois** [fenichelar/ember-simple-auth-token#135]
Our server side authentication expects for parameters for the login instead of JSON.  After authentication everything uses JWT.

Is there anyway to pass username and password initially as form parameters?

**Comment by eluciano11** [comment]
Hi @jgadbois , I think this issue can be solved by extending the authenticator that the add-on provides. Currently we provide `jwt` and `token`. If you're using jwt based authentication you need to import and extend the authenticator. After you extend it you can modify the `makeRequest` function to send the data in form parameters instead of json.

Example:

Create a folder named `authenticators` in you `app`. In it create a js file example: `custom.js`. The file name will be used when you use the `authenticate` function.

``` javascript
//app/authenticators/custom.js

//If you're using jwt you should use the jwt authenticator.
import JWTAuthenticator from 'ember-simple-auth-token/authenticators/jwt';

//If you're not using jwt you should use the token authenticator
//import TokenAuthenticator from 'ember-simple-auth-token/authenticators/token';

export default JWTAuthenticator.extend({
  makeRequest: function(url, data, headers) {
    return Ember.$.ajax({
      url: url,
      data: data,
      type: 'POST',
      beforeSend: (xhr, settings) => {
        xhr.setRequestHeader('Accept', settings.accepts.json);

        if (headers) {
          Object.keys(headers).forEach(key => {
            xhr.setRequestHeader(key, headers[key]);
          });
        }
      }
    });
  }
});

//app/login/route/route.js
import Ember from 'ember';

export default Ember.Route.extend({
  actions: {
    login(data) {
      const controller = this.controllerFor('login');

      controller.set('isLoginIn', true);
      // this.get('session').authenticate('authenticator:nameOfTheCustomAuthenticator', data);
      this.get('session').authenticate('authenticator:custom', data);
    }
  }
});
```

Hope this helps!

**Comment by jgadbois** [comment]
Thanks for following up on this!

On Fri, Mar 25, 2016 at 8:45 AM -0700, "José Padilla" notifications@github.com wrote:

Closed #135.
